doc/getting_started/getting_started_go.rst

```diff
@@ -245,3 +245,207 @@ To send bare Lua code for execution, use ``Eval``:
+
+    *   -   License
+        -   BSD 2-Clause
+        -   MIT
+        -   BSD 2-Clause
+
+    *   -   Last update
+        -   2021
+        -   2021
+        -   2021
+
+    *   -   Known Issues
+        -   `#50 <https://github.com/tarantool/go-tarantool/issues/50>`_,
+            `#89 <https://github.com/tarantool/go-tarantool/issues/89>`_,
+            `#94 <https://github.com/tarantool/go-tarantool/issues/94>`_
+        -   `#41 <https://github.com/viciious/go-tarantool/issues/41>`_
+        -   No known issues
+
+    *   -   Documentation
+        -   README with examples, API described in code comments
+        -   README with examples, code comments
+        -   README with examples
+
+    *   -   Testing / CI / CD
+        -   GitHub Actions
+        -   Travis CI
+        -   GitHub Actions
+
+    *   -   Github Stars
+        -   127
+        -   43
+        -   12
+
+    *   -   Static Analysis
+        -   No
+        -   golint
+        -   golangci-lint
+
+    *   -   Packaging
+        -   go get
+        -   go get
+        -   go get
+
+    *   -   Code coverage
+        -   No
+        -   No
+        -   No
+
+    *   -   msgpack driver
+        -   `vmihailenco/msgpack/v2 <https://github.com/vmihailenco/msgpack/tree/v2>`_
+        -   `tinylib/msgp <https://github.com/tinylib/msgp>`_
+        -   `vmihailenco/msgpack/v5 <https://github.com/vmihailenco/msgpack/tree/v5>`_
+
+    *   -   Async work
+        -   Yes (channels)
+        -   Yes (channels)
+        -   Yes (channels)
+
+    *   -   Batching
+        -   No
+        -   No
+        -   No
+
+    *   -   Schema reload
+        -   Yes (manual pull)
+        -   Yes (manual pull)
+        -   Yes (manual pull)
+
+    *   -   Space / index names
+        -   Yes
+        -   Yes
+        -   Yes
+
+    *   -   Tuples as structures
+        -   Yes (structure and marshall functions must be predefined in Go code)
+        -   No
+        -   Yes (structure and marshall functions must be predefined in Go code)
+
+    *   -   Access tuple fields by names
+        -   Only if marshalled to structure
+        -   No
+        -   Only if marshalled to structure
+
+    *   -   `SQL <https://www.tarantool.io/en/doc/latest/reference/reference_sql/>`_ support
+        -   No (`#62 <https://github.com/tarantool/go-tarantool/issues/62>`_), eval supported
+        -   No (`#18 <https://github.com/viciious/go-tarantool/issues/18>`_, closed), eval supported
+        -   No, eval supported
+
+    *   -   `Interactive transactions <https://www.tarantool.io/en/doc/latest/book/box/stream/>`_
+        -   No (`#101 <https://github.com/tarantool/go-tarantool/issues/101>`_)
+        -   No
+        -   No
+
+    *   -   `Varbinary <https://www.tarantool.io/en/doc/latest/book/box/data_model/>`_ support
+        -   Yes (with in-built language tools)
+        -   Yes (with in-built language tools)
+        -   Questionable (decoding is weird with in-built language tools)
```

Comment:
I would prefer to avoid opinionated words. If you have a link, it is better to give a link.

Reply:
I don't have proper link, it is the result of my own experiments (drafts are stored here https://github.com/DifferentialOrange/tarantool-go-comparison).

Let Tarantool be
```lua
box.cfg{ listen = 3401 }
box.schema.user.grant('guest', 'read,write,execute,create,drop,alter', 'universe')

-- varbinary test

box.schema.space.create('testvarbin')
box.space.testvarbin:create_index('pk', { type = 'tree', parts = {{ field = 1, type = 'varbinary' }} })

buffer = require('buffer')
ffi = require('ffi')

function encode_bin(bytes)
    local tmpbuf = buffer.ibuf()
    local p = tmpbuf:alloc(3 + #bytes)
    p[0] = 0x91
    p[1] = 0xC4
    p[2] = #bytes
    for i, c in pairs(bytes) do
        p[i + 3 - 1] = c
    end
    return tmpbuf
end

function bintuple_insert(space, bytes)
    local tmpbuf = encode_bin(bytes)
    ffi.cdef[[int box_insert(uint32_t space_id, const char *tuple, const char *tuple_end, box_tuple_t **result);]]
    ffi.C.box_insert(space.id, tmpbuf.rpos, tmpbuf.wpos, nil)
end

bintuple_insert(box.space.testvarbin, {0xDE, 0xAD, 0xBE, 0xAF})
```

### tarantool/go-tarantool

```go
package main

import (
	"log"
	"encoding/binary"
	"github.com/tarantool/go-tarantool"
)

func TestVarbinaryTarantool(uri string, user string) {
	opts := tarantool.Opts{User: user}
	conn, err := tarantool.Connect(uri, opts)

	if err != nil {
		log.Fatalf("Connection refused:", err)
	}

	resp, err := conn.Ping()
	log.Println(resp.Code)
	log.Println(resp.Data)
	log.Println(err)

	resp1, err1 := conn.Eval("return box.space.testvarbin:select{}", []interface{}{})

	log.Println("Error", err1)
	log.Println("Code", resp1.Code)
	log.Println("Data", resp1.Data)

	buf := make([]byte, 4)
	binary.BigEndian.PutUint16(buf[0:], 0xa22c)
	binary.BigEndian.PutUint16(buf[2:], 0x04af)

	resp2, err2 := conn.Insert("testvarbin", []interface{}{buf})

	log.Println("Error", err2)
	log.Println("Code", resp2.Code)
	log.Println("Data", resp2.Data)
}
```

```go
package main

func main() {
    var uri string = "127.0.0.1:3401"
    var user string = "guest"

    TestVarbinaryTarantool(uri, user)
}
```

Output is
```
2021/12/20 18:00:13 0
2021/12/20 18:00:13 []
2021/12/20 18:00:13 <nil>
2021/12/20 18:00:13 Error <nil>
2021/12/20 18:00:13 Code 0
2021/12/20 18:00:13 Data [[[[222 173 190 175]]]]
2021/12/20 18:00:13 Error <nil>
2021/12/20 18:00:13 Code 0
2021/12/20 18:00:13 Data [[[162 44 4 175]]]
```

### viciious/go-tarantool

```go
package main

import (
	"context"
	"encoding/binary"
	"log"
	"github.com/viciious/go-tarantool"
)

func TestVarbinaryViciious(uri string, user string) {
	opts := tarantool.Options{User: user}
	conn, err := tarantool.Connect(uri, &opts)

	if err != nil {
		log.Fatalf("Connection refused:", err.Error())
	}

	resp := conn.Exec(context.Background(), &tarantool.Ping{})
	log.Println("Error", resp.Error)
	log.Println("Data", resp.Data)

	expr := "return box.space.testvarbin:select{}"
	q := &tarantool.Eval{Expression: expr}
	resp1 := conn.Exec(context.Background(), q)

	log.Println("Error", resp1.Error)
	log.Println("Data", resp1.Data)

	buf := make([]byte, 4)
	binary.BigEndian.PutUint16(buf[0:], 0xa23c)
	binary.BigEndian.PutUint16(buf[2:], 0x04af)

	query := &tarantool.Insert{Space: "testvarbin", Tuple: []interface{}{buf}}
	resp2 := conn.Exec(context.Background(), query)

	log.Println("Error", resp2.Error)
	log.Println("Data", resp2.Data)
}
```

```go
package main

func main() {
    var uri string = "127.0.0.1:3401"
    var user string = "guest"

    TestVarbinaryViciious(uri, user)
}
```

Output is
```
2021/12/20 18:04:40 Error <nil>
2021/12/20 18:04:40 Data []
2021/12/20 18:04:40 Error <nil>
2021/12/20 18:04:40 Data [[[[222 173 190 175]]]]
2021/12/20 18:04:40 Error <nil>
2021/12/20 18:04:40 Data [[[162 60 4 175]]]
```

### FZambia/tarantool


```go
package main

import (
	"encoding/binary"
	"log"
	"github.com/FZambia/tarantool"
)

func TestVarbinaryFZambia(uri string, user string) {
	opts := tarantool.Opts{User: user}
	conn, errc := tarantool.Connect(uri, opts)
	if errc != nil {
		log.Fatalf("Connection refused: %v", errc)
	}

	resp, err := conn.Exec(tarantool.Ping())

	log.Println("Ping Code", resp.Code)
	log.Println("Ping Data", resp.Data)
	log.Println("Ping Error", err)

	resp1, err1 := conn.Exec(tarantool.Eval("return box.space.testvarbin:select{}", []interface{}{}))

	log.Println("Error", err1)
	log.Println("Code", resp1.Code)
	log.Println("Data", resp1.Data)

	buf := make([]byte, 4)
	binary.BigEndian.PutUint16(buf[0:], 0xa25c)
	binary.BigEndian.PutUint16(buf[2:], 0x04af)

	resp2, err2 := conn.Exec(tarantool.Insert("testvarbin", []interface{}{buf}))

	log.Println("Error", err2)
	log.Println("Code", resp2.Code)
	log.Println("Data", resp2.Data)
}
```

```go
package main

func main() {
    var uri string = "127.0.0.1:3401"
    var user string = "guest"

    TestVarbinaryFZambia(uri, user)
}
```

Output is
```
2021/12/20 18:07:30 Ping Code 0
2021/12/20 18:07:30 Ping Data []
2021/12/20 18:07:30 Ping Error <nil>
2021/12/20 18:07:30 Error <nil>
2021/12/20 18:07:30 Code 0
2021/12/20 18:07:30 Data [[[ޭ��]]]
2021/12/20 18:07:30 Error <nil>
2021/12/20 18:07:30 Code 0
2021/12/20 18:07:30 Data [[�\�]]
```

I'm not sure how to interpret this, but first too seems like they are ok with varbinary and the last one may have some issues.

Reply:
Maybe file an issue and link it from here (just 'questionable', without 'is weird' and so on)? It seems, FZambia's connector trying to decode a varbinary array into a unicode string. Or at least link your test.

Reply:
Removed the comment, filed an issue and added the link to the issue

Reply:
Nice!

Reply:
FZambia replied to me with examples. I have changed the entry to "Yes (decodes to string , see [#6](https://github.com/FZambia/tarantool/issues/6))"

Reply:
Decoding to a Unicode string means that varbinary in not supported properly. If your data in not Unicode you'll occasionally see errors or found the data corrupted (with replacement char) on the client side. So the answer is 'No' in practice.

We can mention that we can do so in a custom decoder into a structure.

Reply:
Changed to `No (decodes to string by default, see [#6](https://github.com/FZambia/tarantool/issues/6>))`

Reply:
Okay so.